EIPS/eip-7934.md

```diff
@@ -0,0 +1,77 @@
+---
+eip: 7934
+title: RLP Execution Block Size Limit
+description: Introduce a protocol-level cap on the maximum RLP-encoded block size to 10 MB, including a 512 KB margin for beacon block size.
+author: Giulio Rebuffo (@Giulio2002), Ben Adams (@benaadams), Storm Slivkoff (@sslivkoff)
+discussions-to: https://ethereum-magicians.org/t/eip-7934-add-bytesize-limit-to-blocks/23589
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-04-16
+---
+
+## Abstract
+
+This proposal introduces a protocol-level cap on the maximum RLP-encoded execution block size to 10 megabytes (MB), which includes a margin of 512 KB to account for beacon block sizes.
+
+## Motivation
+
+Currently, Ethereum does not enforce a strict upper limit on the encoded size of blocks. This lack of constraint can result in:
```

Comment:
I think we should carefully define what is a "block" here. Because for EL we could pick the block header, the block itself (header + block body (so this includes transactions and withdrawals for instance)). Or would it be some serialization of whatever gets sent over `engine_newPayload` (so some kind of RLP serialization of `ExecutionPayloadV_n`?)